absl/algorithm/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+list(APPEND ALGORITHM_PUBLIC_HEADERS
+  "algorithm.h"
+  "container.h"
+)
+
+
+#
+## TESTS
+#
+
+# test algorithm_test
+list(APPEND ALGORITHM_TEST_SRC
+  "algorithm_test.cc"
+  ${ALGORITHM_PUBLIC_HEADERS}
+  ${ALGORITHM_INTERNAL_HEADERS}
+)
+
+absl_header_library(TARGET debugging)
```

Comment:
Target is algorithm

Reply:
I know, I know... I'm waiting the answer related to ALIAS target before taking care of that

Reply:
I'm not sure what you want to do with ALIAS targets. Can you explain in detail (or code) what you want to achieve?

Reply:
@Orphis In short, it is visible in AbseilHelpers.cmake in the function absl_header_library

 - "base", like its name mention it, is the basic component of Abseil,   every other component depends on it.
- In abseil, the include path to specify is the same for every component ( $PROJECT_DIR/absl/ )

-> Every header only component can be a simple target ALIAS on "base", the linker will simply ignore the dependency and still cmake will consider the include path. 
And if one day one-header only target become not header-only anymore, It is enough to just re-define it as a "real" target with a public dependency on "base", and every build script should just continue to run happily. No code generation, no fake symbols.

Reply:
Just create libraries just as listed in the Bazel file please, it will make any transition easier in case any source file are added to those later on.

Also, adding a dummy source file to a library means you can have those listed as targets in your IDE, see the associated files and usually have those headers suggested by autocompletion. `INTERFACE` libraries in CMake don't have any files associated to them, just build flags and are automatically merged with targets linking against them. You can't just use a `STATIC` library for them since some IDEs will error when trying to create an empty static library, some will create unavoidable warnings too.

Reply:
> INTERFACE libraries in CMake don't have any files associated to them, just build flags and are automatically merged with targets linking against them

 I was not speaking about INTERFACE. But that's fine. I suppose the empty file generation will make your life easier if you plan to generate CMake files from Bazel files in future

Reply:
What's concerning to me here, especially in terms of the hypothetical Bazel -> CMake generator, is that there's one target specified here as opposed to the two in the Bazel file -- //absl/algorithm:algorithm and //absl/algorithm:container